extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeCommentsExtractor.java

```diff
@@ -74,45 +73,46 @@ public YoutubeCommentsExtractor(
 
     /**
      * Finds the initial comments token and initializes commentsDisabled.
+     * Also set
      *
      * @return the continuation token or null if none was found
      */
     @Nullable
     private String findInitialCommentsToken() throws ExtractionException {
-
-        final JsonArray jArray = JsonUtils.getArray(nextResponse,
-                "contents.twoColumnWatchNextResults.results.results.contents");
-
-        final Optional<Object> itemSectionRenderer = jArray.stream().filter(o -> {
-            JsonObject jObj = (JsonObject) o;
-
-            if (jObj.has("itemSectionRenderer")) {
-                try {
-                    return JsonUtils.getString(jObj, "itemSectionRenderer.targetId")
-                            .equals("comments-section");
-                } catch (final ParsingException ignored) {
-                }
-            }
-
-            return false;
-        }).findFirst();
-
-        final String token;
-
-        if (itemSectionRenderer.isPresent()) {
-            token = JsonUtils.getString(((JsonObject) itemSectionRenderer.get())
-                            .getObject("itemSectionRenderer").getArray("contents").getObject(0),
-                    "continuationItemRenderer.continuationEndpoint.continuationCommand.token");
-        } else {
-            token = null;
-        }
-
-        if (token == null) {
-            optCommentsDisabled = Optional.of(true);
-            return null;
-        }
-
-        optCommentsDisabled = Optional.of(false);
+        final String token = JsonUtils.getArray(nextResponse,
+                "contents.twoColumnWatchNextResults.results.results.contents")
+                .stream()
+                // Only use JsonObjects
+                .filter(JsonObject.class::isInstance)
+                .map(JsonObject.class::cast)
+                // Only process JsonObjects that have a itemSectionRenderer
+                .filter(jObj -> jObj.has("itemSectionRenderer"))
+                // Check if the comment-section is present
+                .filter(jObj -> {
+                    try {
+                        return "comments-section".equals(
+                                JsonUtils.getString(jObj, "itemSectionRenderer.targetId"));
+                    } catch (final ParsingException ex) {
```

Comment:
Otherwise Android Studio gives warning for unused variable
```suggestion
                    } catch (final ParsingException ignored) {
```

Reply:
FYI: My Android studio shows no warning here:
![grafik](https://user-images.githubusercontent.com/40789489/155012271-9553fcc5-b750-4696-aa36-51b1de09b245.png)